pep-0497-externals.txt

```diff
@@ -0,0 +1,359 @@
+many minutes in to a large compile job.
+
+A key aspect of the usability of this tool is a mechanism to provide more useful
+information to the user about how to obtain those missing resources.
+The proposal is to provide a layer inside of pip which is capable of providing
+information about appropriate package names. It is surprising that vendors
+do not already provide a good tool for this, given that finding appropriate
+header and library files is a common experience in the general linux ecosystem.
+However, it is also a critical aspect of the user experience, and something
+which package maintainers will need to be able to specify precisely. The
+proposed approach would be for pip to supply the ability to install an
+inference layer, usable by users and vendors alike. Usage would be along
+the lines of
+
+    pip install vendor_magic darwin
+
+This would transform the user messages relating to missing dependencies from
+
+    Could not find libatlas.so
+
+to
+
+    Could not find libatlas.so. Run 'port install scipy +atlas'
+
+
+Discussion / FAQ / Issues
+=========================
+
+This section will be updated to capture incoming feedback and represents a
+snapshot of the current concerns.
+
+
+Specification of files (headers, libs, bins) vs packages
+--------------------------------------------------------
+
+There was a discussion about whether specific dependencies (named headers,
+libraries, binaries etc) should be included, or whether general package names
+(such as may be expected by yum, apt, conda) were appropriate. Due to the
+number of package management tools, plus the changing nature of those, it
+was decided that concentrating on the specific files was more useful than
+using package names. It also provides looser coupling between the specification
+of externals and the package management tools.
+
+Location in setup.py rather than requirements.txt
+-------------------------------------------------
+
+Requirements.txt is scoped to the use for local deployments to specify
+requirements precisely for the relevant local environment. Setup.py is the
+appropriate place to specify the information for the package in general.
+
+In future, there may be some scope for including similar directives within
+requirements.txt, but that is out of scope for this PEP.
+
+Prior Approaches
+----------------
+
+Some capacity to support description of external dependencies (among other
+things) was described in PEPs 314 and 345 (Metadata 1.1 and 1.2). These
+describe directives whichs can be placed in the PKG-INFO file. An implementation
+responding to these directives is pending.
+
+This PEP places the directives in setup.py. This approach is closer to the
+typical approach of package maintainers to expressing requirements and places
+the external dependency specification nearer to the Python dependency
+specification. This PEP would supercede PEPs 314 and 345 with respect to
+external dependencies.
```

Comment:
We need to note that it will flow through to PKG-INFO (but the sections I note we need above will include this)